### PR TITLE
improved handling of bottom

### PR DESCRIPTION
**Pull request by andreistefanescu**
not ready for review yet

### CODE REVIEW

**Comment by andreistefanescu** [comment]
Jenkins, test this please

**Comment by andreistefanescu** [comment]
@dwightguth @yilongli please review

File: java-backend/src/main/java/org/kframework/backend/java/util/RewriteEngineUtils.java [hunk 1]
```diff
@@ -66,7 +67,9 @@ public static boolean isSubsortedEq(Term big, Term small, TermContext context) {
                     Term evalLookupOrChoice = evaluateLookupOrChoice(lookupOrChoice, crntSubst, context);
 
                     boolean resolved = false;
-                    if (evalLookupOrChoice == null || evalLookupOrChoice instanceof Bottom) {
+                    if (evalLookupOrChoice instanceof Bottom
+                            || DataStructures.isLookup(evalLookupOrChoice)
```
Comment (yilongli) [inline java-backend/src/main/java/org/kframework/backend/java/util/RewriteEngineUtils.java]: add an `isLookupOrChoice`?

Comment (yilongli) [inline java-backend/src/main/java/org/kframework/backend/java/util/RewriteEngineUtils.java]: what happens if `evalLookupOrChoice` is `null`?

File: java-backend/src/main/java/org/kframework/backend/java/kil/DataStructures.java [hunk 2]
```diff
@@ -33,6 +33,16 @@ static KItem lookup(Term base, Term key, TermContext context) {
         return KItem.of(klabel, kList, context);
     }
 
+    static boolean isChoice(Term term) {
+        return term instanceof KItem
+                && ((KItem) term).kLabel() instanceof KLabelConstant
+                && (((KItem) term).kLabel().toString().equals("Map:choice")
```
Comment (yilongli) [inline java-backend/src/main/java/org/kframework/backend/java/kil/DataStructures.java]: extract `"Map:choice"` to be a constant?

File: java-backend/src/main/java/org/kframework/backend/java/kil/DataStructures.java [hunk 3]
```diff
@@ -33,6 +33,16 @@ static KItem lookup(Term base, Term key, TermContext context) {
...
+                        || ((KItem) term).kLabel().toString().equals("Set:choice"))
```
Comment (yilongli) [inline java-backend/src/main/java/org/kframework/backend/java/kil/DataStructures.java]: same as above

**Comment by yilongli** [comment]
lgtm

File: java-backend/src/main/java/org/kframework/backend/java/util/RewriteEngineUtils.java (hunk 1 above)
Comment (andreistefanescu) [inline java-backend/src/main/java/org/kframework/backend/java/util/RewriteEngineUtils.java]: `evaluateLookupOrChoice` cannot return null right now. I can add it back if you wish.

Comment (yilongli) [inline java-backend/src/main/java/org/kframework/backend/java/util/RewriteEngineUtils.java]: no, it's good

**Comment by dwightguth** [comment]
Actually, can you put the test in tests/regression/regression.k to avoid an extra call to kompile? Otherwise lgtm

**Pull request merged**